roles/matrix-bot-matrix-registration-bot/templates/systemd/matrix-bot-matrix-registration-bot.service.j2

```diff
@@ -0,0 +1,33 @@
+#jinja2: lstrip_blocks: "True"
+[Unit]
+Description=Matrix registration bot
+{% for service in matrix_bot_matrix_registration_bot_systemd_required_services_list %}
+Requires={{ service }}
+After={{ service }}
+{% endfor %}
+{% for service in matrix_bot_matrix_registration_bot_systemd_wanted_services_list %}
+Wants={{ service }}
+{% endfor %}
+DefaultDependencies=no
+
+[Service]
+Type=simple
+Environment="HOME={{ matrix_systemd_unit_home_path }}"
+ExecStartPre=-{{ matrix_host_command_sh }} -c '{{ matrix_host_command_docker }} kill matrix-bot-matrix-registration-bot 2>/dev/null || true'
+ExecStartPre=-{{ matrix_host_command_sh }} -c '{{ matrix_host_command_docker }} rm matrix-bot-matrix-registration-bot 2>/dev/null || true'
+
+ExecStart={{ matrix_host_command_docker }} run --rm --name matrix-bot-matrix-registration-bot \
+			--log-driver=none \
+			--cap-drop=ALL \
```

Comment:
Can we add `--read-only` for additional security here or does the container choke with that?

Reply:
A simple `--read-only`does not work (`matrix-bot-matrix-registration-bot[1870772]: OSError: [Errno 30] Read-only file system: 'config.toml'` but I think I will rewrite configuration bit of the role to mount the config directory. This has the advantage that the `config.toml` stays accessible to the user.

Reply:
We're configuring `matrix-registration-bot` using environment variables though, not via `config.yml`. If the bot can be fully-configured using environment varialbes only, we'd better refrain from using `config.yml` at all.

Seems like your `matrix-registration-bot` code tries to write some configuration on startup? Perhaps that could be reworked?

Reply:
The bot can only partially be configured via `config.yml` as allowed users etc... live in `config.toml` which is provided by the underlying `simplematrixbotlib`. As this file has to be persistent and users may want to modify is manually I would add `config.yml` and `config.toml` to the config folder (and get rid of the environment variables). The bot furthermore writes a `session.txt` (which can be ephemeral, but it is better if it's not).

Reply:
Should we switch away from environment variables to `config.yml` then?

As for the `session.txt` file (or other persistent data managed by the bot itself), it should probably go into a `/data` directory that is mounted separately.

The `config/` directory (or `config/config.yml`) can be mounted into the container as read-only, while the `data/` directory is mounted writable (no read-only modifier).

Reply:
Yes, I think that would be good for now. I will have a look at the reminder bot that uses a similar structure.